crypto/include/aes_gcm.h

```diff
@@ -64,6 +64,35 @@ typedef struct {
 
 #endif /* OPENSSL */
 
+#ifdef WOLFSSL
+#define MAX_AD_SIZE 2048
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+#ifndef WOLFSSL_USER_SETTINGS
+#include <wolfssl/options.h>
+#endif
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/aes.h>
+
+typedef struct {
+    int key_size;
+    int tag_len;
+    int aad_size;
+#ifndef WOLFSSL_AESGCM_STREAM
```

Comment:
Which of the two api's is newer/preferred and do you really need to support both ?

Reply:
Prefer streaming but wolfSSL can be compiled without.
Some platforms don't have assembly code for streaming.